### PR TITLE
Some more tweaks enabled in dev11

### PR DESCRIPTION
**Pull request by chrisbanes**

### CODE REVIEW

File: common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        CircularProgressIndicator(
+            strokeWidth = (diameter * strokePerDiameterDp).coerceAtLeast(1.dp),
+            color = color
+        )
+    }
+}
+
+// Default stroke size
+private val defaultStrokeWidth = 4.dp
```
Comment (nickbutcher) [inline common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt:51]: I've been using PascalCase for constants like the material libs.

File: common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import androidx.ui.unit.min
+
+@Composable
+fun AutoSizedCircularProgressIndicator(
+    modifier: Modifier = Modifier,
+    color: Color = MaterialTheme.colors.primary
+) {
+    WithConstraints(modifier) {
+        val diameter = with(DensityAmbient.current) {
+            min(constraints.maxWidth, constraints.maxHeight).toDp() - defaultPadding
```
Comment (nickbutcher) [inline common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt:40]: Hoist the padding?

File: common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+// Default stroke size
+private val defaultStrokeWidth = 4.dp
+// Preferred diameter for CircularProgressIndicator
+private val defaultDiameter = 40.dp
+// Default padding added by CircularProgressIndicator
+private val defaultPadding = 4.dp
+
+private val strokePerDiameterDp = defaultStrokeWidth / defaultDiameter
```
Comment (nickbutcher) [inline common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt:57]: Don't love the name. What about `strokeFraction`?

File: common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt (hunk 2 above)
Comment (chrisbanes) [reply common-ui-compose/src/main/java/app/tivi/common/compose/AutoSizedCircularProgressIndicator.kt:40]: The padding is an internal constant in `CircularProgressIndicator` which I need to take into account to find the 'true' diameter.. Not much point hoisting it as it's not really a parameter imo.

File: common-ui-compose/src/main/java/app/tivi/common/compose/gradientscrim.kt [hunk 4]
```diff
@@ -17,46 +17,56 @@
...
             val x = i * 1f / (numStops - 1)
-            val opacity = x.pow(3.0f)
+            val opacity = x.pow(decay)
             color.copy(alpha = baseAlpha * opacity)
         }
     }
 
+    var height by state { 0f }
+
+    val shader = remember(height) {
```
Comment (nickbutcher) [inline common-ui-compose/src/main/java/app/tivi/common/compose/gradientscrim.kt:59]: Also based on `colors`
```suggestion
    val shader = remember(colors, height) {
```

**Pull request merged**